include/zephyr/drivers/sensor_v2.h

```diff
@@ -0,0 +1,931 @@
+
+/**
+ * @typedef sensor_read_uint64_data_t
+ * @brief Sensor API function for reading a uint64_t data point
+ *
+ * @see sensor_read_uint64_data() for argument description
+ */
+typedef int (*sensor_read_uint64_data_t)(const struct device *sensor, uint8_t sensor_type,
+					 struct sensor_uint64_data *data);
+
+/**
+ * @typedef sensor_read_temperature_t
+ * @brief Sensor API function for reading the sensor's internal temperature
+ *
+ * @see sensor_read_temperature() for argument description
+ */
+typedef int (*sensor_read_temperature_t)(const struct device *sensor, uint8_t sensor_type,
+					 int8_t *temperature_c);
+
+/**
+ * @typedef sensor_get_range_t
+ * @brief Sensor API function for getting the current range
+ *
+ * @see sensor_get_range() for argument description
+ */
+typedef int (*sensor_get_range_t)(const struct device *sensor, uint8_t sensor_type,
+				  uint32_t *range);
+
+/**
+ * @typedef sensor_set_range_t
+ * @brief Sensor API function for setting the current range
+ *
+ * @see sensor_set_range() for argument description
+ */
+typedef int (*sensor_set_range_t)(const struct device *sensor, uint8_t sensor_type, uint32_t range,
+				  bool round_up);
+
+/**
+ * @typedef sensor_get_three_axis_bias_t
+ * @brief Sensor API function for getting the current bias and the temperature at which it was set
+ *
+ * @see sensor_get_three_axis_bias() for argument description
+ */
+typedef int (*sensor_get_three_axis_bias_t)(const struct device *sensor, uint8_t sensor_type,
+					    int8_t *temperature, fp_t *bias_x, fp_t *bias_y,
+					    fp_t *bias_z);
+
+/**
+ * @typedef sensor_set_three_axis_bias_t
+ * @brief Sensor API function for setting the bias and the temperature at which it was attained
+ *
+ * @see sensor_set_three_axis_bias() for argument description
+ */
+typedef int (*sensor_set_three_axis_bias_t)(const struct device *sensor, uint8_t sensor_type,
+					    int8_t temperature, fp_t bias_x, fp_t bias_y,
+					    fp_t bias_z);
+
+/**
+ * @typedef sensor_get_resolution_t
+ * @brief Sensor API function for getting the current resolution in bits/sample
+ *
+ * @see sensor_get_resolution() for argument description
+ */
+typedef int (*sensor_get_resolution_t)(const struct device *sensor, uint8_t sensor_type,
+				       uint8_t *resolution);
+
+/**
+ * @typedef sensor_set_resolution_t
+ * @brief Sensor API function for setting the current resolution in bits/sample
+ *
+ * @see sensor_set_resolution() for argument description
+ */
+typedef int (*sensor_set_resolution_t)(const struct device *sensor, uint8_t sensor_type,
+				       uint8_t resolution, bool round_up);
+
+/**
+ * @typedef sensor_process_data_callback_t
+ * @brief Callback type for handling asynchronous data
+ *
+ * @param sensor Pointer to the sensor device
+ * @param sensor_type The type of sensor that generated the data
+ * @param data The data generated, it should be cast to one of the sensor_*_data structs based on
+ *        the sensor_type.
```

Comment:
Can this be more than one sample if the FIFO is enabled?

Reply:
You'll get multiple callbacks here if FIFO is enabled. I'll update the doc.